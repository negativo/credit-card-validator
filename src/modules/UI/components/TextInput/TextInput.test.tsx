import React from 'react';
import { TextInput } from './';
import renderer from 'react-test-renderer';

it('renders correctly with props', () => {
  const tree = renderer.create(<TextInput>TextInput Test</TextInput>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with Label', () => {
  const tree = renderer
    .create(<TextInput label="test label">TextInput Test</TextInput>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with props', () => {
  const tree = renderer
    .create(
      <TextInput
        isValid={true}
        label="test props"
        onChange={() => null}
        value="test value"
      >
        TextInput Test
      </TextInput>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
