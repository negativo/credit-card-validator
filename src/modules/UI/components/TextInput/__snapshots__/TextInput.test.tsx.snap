// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Label 1`] = `
<div
  className="sc-AxirZ fSNuUs"
>
  <div
    className="sc-AxjAm eMwonu"
    type="primary"
  >
    test label
  </div>
  <input
    className="sc-AxiKw ckEfbS"
    type="text"
  />
</div>
`;

exports[`renders correctly with props 1`] = `
<div
  className="sc-AxirZ fSNuUs"
>
  <div
    className="sc-AxjAm eMwonu"
    type="primary"
  />
  <input
    className="sc-AxiKw ckEfbS"
    type="text"
  />
</div>
`;

exports[`renders correctly with props 2`] = `
<div
  className="sc-AxirZ fSNuUs"
>
  <div
    className="sc-AxjAm eMwonu"
    type="primary"
  >
    test props
  </div>
  <input
    className="sc-AxiKw gMeQUp"
    onChange={[Function]}
    type="text"
    value="test value"
  />
</div>
`;
